### PR TITLE
EPMRPP-45431 || System attributes retrieving

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class SystemAttributesExtractor {
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: Please cover this class with unit-tests

Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: done in https://github.com/reportportal/client-java/pull/65/commits/da4341a1774b0e047a93164ea33b7981f8544ae0

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	private static final Logger LOGGER = LoggerFactory.getLogger(SystemAttributesExtractor.class);
+
+	private static final String ATTRIBUTE_VALUE_SEPARATOR = "|";
+
+	private SystemAttributesExtractor() {
+		//static only
+	}
+
+	public static Set<ItemAttributesRQ> extract(final String resource) {
+		Properties properties = new Properties();
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: Can you explain in few words how do these system attributes work?

Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: There are 2 types:
1) from a `property` file - for now it's only `agent name` and `agent version`
2) from the `System` - OS name, arch and version 
All key/value are parsed from the properties and converted to `ItemAttributesRQ` with `system` field equal to `true`

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	 *
+	 * @param resource Path to the resource in classpath
+	 * @return {@link Set} of {@link ItemAttributesRQ}
+	 */
+	public static Set<ItemAttributesRQ> extract(final String resource) {
+		Properties properties = new Properties();
+
+		ofNullable(resource).flatMap(res -> ofNullable(SystemAttributesExtractor.class.getClassLoader().getResource(res)))
+				.ifPresent(url -> {
+					try (InputStreamReader inputStreamReader = new InputStreamReader(url.openStream(), StandardCharsets.UTF_8)) {
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: can be moved to the separate method with input stream and property as parameters

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+			if (file.exists()) {
+				try (InputStreamReader inputStreamReader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)) {
+					properties.load(inputStreamReader);
+				} catch (IOException e) {
+					LOGGER.warn("Unable to load system properties file");
+				}
+			}
+		});
+
+		return getAttributes(properties);
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: refactor to make it more clear and uderstandable
probably divide into separate methods

File: src/test/java/com/epam/reportportal/utils/properties/SystemAttributesExtractorTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.regex.Pattern;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class SystemAttributesExtractorTest {
+
+	private static final Map<String, Pattern> agentProperties = new HashMap<>();
```
Comment (pbortnik) [inline src/test/java/com/epam/reportportal/utils/properties/SystemAttributesExtractorTest.java]: do we really need these two maps here?

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+		if (!values.isEmpty()) {
+			return Optional.of(new ItemAttributesRQ(attributeKey, StringUtils.join(values, ATTRIBUTE_VALUE_SEPARATOR), true));
+		} else {
+			return Optional.empty();
+		}
+	}
+
+	private enum DefaultProperties {
+		OS("os", true, "os.name", "os.arch", "os.version"),
+		AGENT("agent", false, "agent.name", "agent.version");
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: What about jvm info?

Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: JVM properties provided

File: src/test/java/com/epam/reportportal/utils/properties/SystemAttributesExtractorTest.java (hunk 5 above)
Comment (Yumfriez) [reply src/test/java/com/epam/reportportal/utils/properties/SystemAttributesExtractorTest.java]: fixed

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: updated

File: src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java]: Stream opening requires IOException handling, so it won't be less complicated after moving in the separate method

**Pull request merged**